Modules/_functoolsmodule.c

```diff
@@ -6,6 +6,7 @@
 #include "pycore_object.h"        // _PyObject_GC_TRACK
 #include "pycore_pystate.h"       // _PyThreadState_GET()
 #include "pycore_tuple.h"         // _PyTuple_ITEMS()
+#include "pycore_critical_section.h"  // Py_BEGIN_CRITICAL_SECTION
```

Comment:
The includes are mostly sorted. Would you please put this between pycore_call.h and pycore_dict.h?